src/main/java/com/cloudbees/jenkins/GitHubPushTrigger.java

```diff
@@ -299,13 +301,17 @@ public FormValidation doCheckHookUrl(@QueryParameter String value) {
                     return FormValidation.warning("It doesn't look like " + value + " is talking to any Jenkins. Are you running your own app?");
                 }
                 RSAPublicKey key = identity.getPublic();
-                String expected = new String(Base64.encodeBase64(key.getEncoded()));
+                String expected = new String(Base64.encodeBase64(key.getEncoded()), "UTF-8" );
                 if (!expected.equals(v)) {
                     // if it responds but with a different ID, that's more likely wrong than correct
                     return FormValidation.error(value+" is connecting to different Jenkins instances");
                 }
 
                 return FormValidation.ok();
+            } catch (UnsupportedEncodingException ex) {
+                // Should not happen
+                // TODO: get rid of it after migrating to JDK7
+                return FormValidation.error("Unsupported exception "+ex.getLocalizedMessage());
```

Comment:
I'll add spaces. Formatting is an overkill for a single string addition to the end